R/target_encoder.R

```diff
@@ -0,0 +1,34 @@
+#' target_encoder
+#'
+#' This function encodes categorical variables with average target values for each category.
+#'
+#' @param X_train A tibble representing the training data set containing some categorical features/columns.
+#' @param X_test A tibble representing the test set, containing some set of categorical features/columns.
+#' @param y A numeric vector or character vector representing the target variable. 
+#'  If the objective is "binary", then the vector should only contain two unique values.
```

Comment:
Maybe mention here that the positive class should be encoded as 1, and the negative as 0 or something? What do you think?

Reply:
I think this might depend on the input. The encoder would just calculate the average regardless of the meaning of targets

Reply:
Ah, okay that makes sense.